simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -1547,70 +1547,52 @@ auto HdMapUtils::getLongitudinalDistance(
-      return next_lanelet_itr == next_lanelet_ids.end();
-    } else {
-      return false;
-    }
+
+  const auto with_lane_change =
+    [this](const lanelet::Id current_lanelet, const lanelet::Id next_lanelet) -> bool {
+    const auto next_lanelet_ids = getNextLaneletIds(current_lanelet);
+    return std::none_of(
+      next_lanelet_ids.begin(), next_lanelet_ids.end(),
+      [next_lanelet](const lanelet::Id id) { return id == next_lanelet; });
   };
 
-  /// @note in this for loop, some cases are marked by @note command. each case is explained in the document.
   /// @sa https://tier4.github.io/scenario_simulator_v2-docs/developer_guide/DistanceCalculation/
-  for (unsigned int i = 0; i < route.size(); i++) {
-    if (i < route.size() - 1 && with_lane_change(allow_lane_change, route[i], route[i + 1])) {
-      /// @note "the lanelet before the lane change" case
-      traffic_simulator_msgs::msg::LaneletPose next_lanelet_pose;
-      next_lanelet_pose.lanelet_id = route[i + 1];
-      next_lanelet_pose.s = 0.0;
-      next_lanelet_pose.offset = 0.0;
-
-      if (
-        auto next_lanelet_origin_from_current_lanelet =
-          toLaneletPose(toMapPose(next_lanelet_pose).pose, route[i], 10.0)) {
-        distance += next_lanelet_origin_from_current_lanelet->s;
+  double accumulated_distance = 0.0;
+  for (std::size_t i = 0UL; i < route.size() - 1UL; ++i) {
+    if (with_lane_change(route[i], route[i + 1UL])) {
+      const auto curr_lanelet_spline = getCenterPointsSpline(route[i]);
+      const auto next_lanelet_spline = getCenterPointsSpline(route[i + 1UL]);
+
+      constexpr double lanelet_starting_s = 0.0;
+      const double lanelet_half_min_s =
+        std::min(curr_lanelet_spline->getLength(), next_lanelet_spline->getLength()) * 0.5;
+
+      constexpr double matching_distance = 10.0;
+      const auto curr_start_matching_s = curr_lanelet_spline->getSValue(
+        next_lanelet_spline->getPose(lanelet_starting_s), matching_distance);
+      const auto next_start_matching_s = next_lanelet_spline->getSValue(
+        curr_lanelet_spline->getPose(lanelet_starting_s), matching_distance);
+      const auto curr_middle_matching_s = curr_lanelet_spline->getSValue(
+        next_lanelet_spline->getPose(lanelet_half_min_s), matching_distance);
+      const auto next_middle_matching_s = next_lanelet_spline->getSValue(
+        curr_lanelet_spline->getPose(lanelet_half_min_s), matching_distance);
+
+      if (curr_start_matching_s.has_value()) {
+        accumulated_distance += curr_start_matching_s.value();
+      } else if (next_start_matching_s.has_value()) {
+        accumulated_distance -= next_start_matching_s.value();
+      } else if (curr_middle_matching_s.has_value()) {
+        accumulated_distance += curr_middle_matching_s.value() - lanelet_half_min_s;
+      } else if (next_middle_matching_s.has_value()) {
+        accumulated_distance -= next_middle_matching_s.value() - lanelet_half_min_s;
```

Comment:
is this approach explained in the doc? it is worth adding a few words of clarification, after all

Reply:
I have updated the decumentation